### PR TITLE
Allow blocks taperecall for any Tier if inputblocks provided

### PR DESCRIPTION
**Pull request by novicecpp**
Fix https://github.com/dmwm/CRABServer/issues/7528

I have changed what we agree in https://github.com/dmwm/CRABServer/issues/7528#issuecomment-1425955091,
from **Allow to recall blocks in the new list of dataTier (e.g. RAW, RECO)**,
to **Allow to recall any number of blocks if it not in `tiersToRecall`  as long as 1) `inputblocks` provided 2) not exceed the limit**.
(The limitation can config via `config.TaskWorker.maxTierToBlockRecallSizeTB`)

This way there is no need to maintain possible dataTier and it make if-else of taperecall's decision more simpler.

Also, simply remove [DBSDataDiscovery.py#L453-L460](https://github.com/dmwm/CRABServer/blob/9e822e562a3200c2eacebb5f93d66ca2b17d3583/src/python/TaskWorker/Actions/DBSDataDiscovery.py#L453-L460) and let it fall through the if-else below.

### CODE REVIEW

File: src/python/TaskWorker/Actions/DBSDataDiscovery.py [hunk 1]
```diff
@@ -254,6 +252,24 @@ def requestTapeRecall(self, blockList=[], system='Dynamo', msgHead=''):   # pyli
...
+    def getBlocksSizeBytes(self, dataset, blocks=None):
+        """
+        Get block size of dataset from DBS and return the total size of all blocks or blocks in `blocks` argument.
+        """
+        self.dbs.checkDatasetPath(dataset)
+        args = {'dataset': dataset, 'detail': True}
+        try:
+            blocksSummaries = self.dbs.dbs.listBlockSummaries(**args)
+        except dbsClientException as ex:
+            msg = "Error in DBSReader.listFileBlocks(%s)\n" % dataset
```
Comment (belforte) [inline src/python/TaskWorker/Actions/DBSDataDiscovery.py:264]: change to "Error in DBSReader.listBlockSummaries|

**Review by belforte — COMMENTED** [review]
I like it very much thjis new way of doing things, thanks !
I have two general comments:
1. some messages to isers could be slightly improved, I can do that later
2. I do not like that we now compute the size of the data to be recalled twice, in the main code calling DBS blockSummary API, and in requeatTapeRecall calling Rucio client. Once should be enough ! But also I need to check why I did not use DBS blocksummary to begin with, if it was due to "effort" in referring to DBS client inside tape recall, or if it was beacuse it is an expensive call for DBS vs. a cheap one in Rucio. I.e. if blockSummaries does (much) more than we need. Or maybe simply you figured out how to call DBS API "bypassing" DBSReader, whjile I did not !

File: src/python/TaskWorker/Actions/DBSDataDiscovery.py [hunk 2]
```diff
@@ -92,7 +89,7 @@ def checkBlocksSize(self, blocks):
                 msg += "\nhttps://twiki.cern.ch/twiki/bin/view/CMSPublic/CRAB3FAQ"
                 raise TaskWorkerException(msg)
 
-    def requestTapeRecall(self, blockList=[], system='Dynamo', msgHead=''):   # pylint: disable=W0102
+    def requestTapeRecall(self, blockList, sizeToRecall, system='Dynamo', msgHead=''):   # pylint: disable=W0102
         """
         :param blockList: a list of blocks to recall from Tape to Disk
```
Comment (belforte) [inline src/python/TaskWorker/Actions/DBSDataDiscovery.py:94]: add a description of new sizeToRecall param
Also time to change the default for system from Dynamo to Rucio

File: src/python/TaskWorker/Actions/DBSDataDiscovery.py [hunk 3]
```diff
@@ -372,6 +366,8 @@ def executeInternal(self, *args, **kwargs):
                 msg = "No locations found with Rucio for this dataset"
                 self.logger.warning(msg)
 
+        self.logger.debug("Dataset size in bytes: %s", totalSizeBytes)
```
Comment (belforte) [inline src/python/TaskWorker/Actions/DBSDataDiscovery.py:369]: convert to GB or TB in the message.

**Comment by belforte** [comment]
I am done reviewing. Looks better and better :-)

**Review by belforte — APPROVED** [review]

**Comment by belforte** [comment]
feel free to merge

**Comment by novicecpp** [comment]
Thanks Stefano.
You review/comment is faster than pylint jobs and my response :)

**Pull request merged**